.circleci/config.yml

```diff
@@ -35,10 +35,45 @@ commands:
             - run:
                   name: prepare test git user
                   command: git config --global user.email "circleci@example.com" && git config --global user.name "CircleCi Build"
+    run-regressions:
+        parameters:
+            force_update_regression_cache:
+                type: boolean
+                default: false
+            regression_color:
+                type: string
+            regression_scale:
+                type: string
+        steps:
+            - restore_cache:
+                  keys:
+                      - v1-golden-images-<< parameters.regression_color >>-<< parameters.regression_scale >>-b9e157e40b4274683707eccf7f8d400ed4c719bc
+                      - v1-golden-images-master-<< parameters.regression_color >>-<< parameters.regression_scale >>-
+            - run: yarn test:visual:ci --color=<< parameters.regression_color >> --scale=<< parameters.regression_scale >> || << parameters.force_update_regression_cache >>
+            - run: cp -RT test/visual/screenshots-current/ci test/visual/screenshots-baseline/ci
```

Comment:
could this be moved into the force update regression cache step?

Reply:
I was having some trouble with this, but I'll experiment a little more, particularly with your ideas around `on_success` and `on_fail`.